Make worker-Nodes take preference in nncp ci_kustomize

evallesp opened the pull request:
As for example in 3+3 scenarios, we want to run all openstack namespaces pods
in the worker nodes.

As we have different type of nodes: ocp, crc and ocp-worker; we want to
select the nodes by taking precedende to worker nodes if defined.
If not, we want to check if it's a SNO crc deployment, and lastly, we'll
go by typical scenarios where the control-plane, master and workers in the same
nodes.

evallesp commented:
recheck

sdatko reviewed (dismissed):
LGTM

karelyatin commented on roles/ci_gen_kustomize_values/templates/bmo01/network-values/values.yaml.j2:
```diff
@@ -2,9 +2,17 @@
 {% set ns = namespace(ocp_index=0) %}
+
+{% if cifmw_networking_env_definition.instances.keys() | select('match', '^ocp-worker') | list | length > 0 %}
+{% set filter="^ocp-worker.*" %}
```
^ocp-worker.* -> ^ocp-worker for consistency with other filters

evallesp replied:
Done, thanks!

karelyatin reviewed (approved):
/lgtm

evallesp commented:
@imatza-rh @eurijon @openstack-k8s-operators/bgp  @openstack-k8s-operators/dcn Could you review?

tosky commented:
Just out of curiosity and to improve my knowledge: we have some cases (see some DTs in architecture.git) where we mark the worker nodes as not schedulable (maybe because it is used for testing purposes) and all masters are used also for control-plane. Would this change affect those scenarios?

evallesp commented:
Moving to DNM for supporting scenarios where worker node is set as non-scheludable.

evallesp commented:
@tosky After talk with @eduolivares it seem we need to stop configuring BGP in master nodes. Its kustomization + this patch makes the BGP jobs fail. I've removed BGP template nodes from this patch and we'll include them once we've removed master nodes BGP configuration.

@eduolivares @imatza-rh @sdatko @danpawlik @bshewale @amartyasinha Could you re-review again?
Thanks a lot.

evallesp commented:
> Hope you verify that with other promotion jobs.

Moving back DNM, thanks for the insight, creating test project with the whole jobs instead of picking a subgroup of them.

evallesp commented:
BGP scenarios PR: https://github.com/openstack-k8s-operators/ci-framework/pull/3337

fultonj commented:
FWIW I'm +1 on this change

evallesp commented:
unialpha01 testproject were green: https://sf.apps.int.gpc.ocp-hub.prod.psi.redhat.com/zuul/t/components-integration/build/b0ab84dbde56442faa1b3985d08ec13c

fultonj reviewed (approved):
/approve
/lgtm
